platform/internal/mbed_fault_handler.h

```diff
@@ -48,24 +52,55 @@ typedef struct {
     uint32_t EXC_RETURN;
     uint32_t CONTROL;
 } mbed_fault_context_t;
+#elif defined TARGET_CORTEX_A
+// This is not currently used, but would be a plausible implementation
+typedef struct {
+    uint32_t R0_reg;
+    uint32_t R1_reg;
+    uint32_t R2_reg;
+    uint32_t R3_reg;
+    uint32_t R4_reg;
+    uint32_t R5_reg;
+    uint32_t R6_reg;
+    uint32_t R7_reg;
+    uint32_t R8_reg;
+    uint32_t R9_reg;
+    uint32_t R10_reg;
+    uint32_t R11_reg;
+    uint32_t R12_reg;
+    uint32_t SP_reg;
+    uint32_t LR_reg;
+    uint32_t PC_reg;
+    uint32_t CPSR;
+    uint32_t SP_usr;
+    uint32_t LR_usr;
+} mbed_fault_context_t;
+#else
+// Dummy for mbed_crash_data_t
+typedef struct {
+} mbed_fault_context_t;
+#endif
+
 
 //Fault type definitions
+#ifdef TARGET_CORTEX_M
 //WARNING: DO NOT CHANGE THESE VALUES WITHOUT MAKING CORRESPONDING CHANGES in except.S files.
 #define HARD_FAULT_EXCEPTION       (0x10) //Keep some gap between values for any future insertion/expansion
 #define MEMMANAGE_FAULT_EXCEPTION  (0x20)
 #define BUS_FAULT_EXCEPTION        (0x30)
 #define USAGE_FAULT_EXCEPTION      (0x40)
+#endif
 
 //This is a handler function called from Fault handler to print the error information out.
 //This runs in fault context and uses special functions(defined in mbed_fault_handler.c) to print the information without using C-lib support.
-void mbed_fault_handler(uint32_t fault_type, const mbed_fault_context_t *mbed_fault_context_in);
+MBED_NORETURN void mbed_fault_handler(uint32_t fault_type, const mbed_fault_context_t *mbed_fault_context_in);
```

Comment:
Can we make the fault handlers to be marked as no return only in release mode ?
Or at least provide a dead `bx lr` only present when not building in release profile ?
Returning from precise faults has multiple times proven extremely useful.
This is nowadays rather cumbersome to achieve in mbed and I'd like to see it become easier rather than marking more functions as no return.

Reply:
This would take more than undoing this change - the assembler glue currently doesn't support `mbed_fault_handler` returning, so this was just matching the C annotation to the assembler use. (Get a warning if we do try to return).

Are you thinking more of doing a `BX lr` directly from the fault handler vector to unwind the stack to overcome backtrace unwinding problems in debuggers?

Reply:
Yes. I often break on `HardFault_Handler` and try to find a `bx lr` somewhere in the flash to then 
```
set $pc = 0x_____
stepi
```
this is most on the time much easier to debug/understand the context of the fault than hopping for the debugger to sort out the content of a potentially corrupted memory.